README.md

```diff
@@ -48,33 +65,470 @@ cache read/white process, etc. Command handler can be sync and publish the
 started and is in the pending state at the moment.
-- **Data** which holds the output produced by the command execution
-- **Error** notifies that an error has been occurred during the command
-execution
+- `Data` event holds the output produced by the command execution.
+- `Error` notifies that an error has been occurred during the command
+execution.
 
 ### Reaction
 
 **Reaction** responds to event published by command handler. It is supposed
 to handle 3 possible event types by describing the desired behavior in the
 each scenario:
 
-- **Wait** function reacts on `Progress` type of the event
-- **Consume** function reacts on `Data` type of the event.
-- **Rescue** function is a fallback for the case when `Error` type of the event
-has been received.
+- `Wait` function reacts on `Progress` type of the event
+- `Consume` function reacts on `Data` type of the event.
+- `Rescue` function is a fallback for the case when `Error` event has been
+received.
 
 Normally reaction performs UI updates, but could also be used for other kinds
 of output processing.
 
+## The flow
+
+### Command execution
+
+The first step is to declare a command. Your command type has to conform to
+`Aftermath.Command` protocol and the `Output` type must be implicitly specified.
+
+Let's say we want to fetch a list of books from some untrusted resource and
+correct typos in titles and author names 🤓.
+
+```swift
+// This is our model we are going to work with.
+struct Book {
+  let id: Int
+  let title: String
+  let author: String
+}
+
+struct BookListCommand: Command {
+  // Result of this command will be a list of books.
+  typealias Output = [Book]
+}
+
+struct BookUpdateCommand: Command {
+  // Result of this command will be an updated book.
+  typealias Output = Book
+
+  // Let's pass the entire model to the command to simplify this example.
+  // Ideally we wouldn't do that because a command is supposed to be as simple
+  // as possible, only with attributes that are needed for handler.
+  let book: Book
+}
+```
+
+**Note** that any type can play a role of `Output`, so if we want to add a date
+to our `BookUpdateCommand` it could look like the following:
+
+```swift
+typealias Output = (Book, NSDate)
+```
+
+In order to execute a command you have to conform to `CommandProducer` protocol:
+
+```swift
+class ViewController: UITableViewController, CommandProducer {
+
+  // Fetch a list of books.
+  func load() {
+    execute(command: BookListCommand())
+  }
+
+  // Update a single book with corrected title and/or author name.
+  func update(book: Book) {
+    execute(command: BookUpdateCommand(book: book))
+  }
+}
+```
+
+### Command handling
+
+Command is an intention that needs to be translated into action by handler.
+Command handler is responsible for publishing events to notify about results of
+the operation it performs. Command handler type has to conform to
+`Aftermath.CommandHandler` protocol that needs to know about the command type
+it will work with:
```

Comment:
> Command handler is responsible for publishing events to notify about results of
the operation it performs. Command handler type has to conform to
`Aftermath.CommandHandler` protocol that needs to know about the command type
it will work with:

> The command handler is responsible for publishing events to notify about results of
the operation it performs. The command handlers type has to conform to
`Aftermath.CommandHandler` protocol, that needs to know about the command type
it will work with: